classes/app.rb

```diff
@@ -0,0 +1,93 @@
+require_relative 'book'
+require_relative 'person'
+require_relative 'student'
+require_relative 'teacher'
+require_relative 'rental'
+
+class App
+  def initialize
+    @books = []
+    @people = []
+    @rentals = []
+    @students = []
+  end
+
+  def list_books
+    @books.each_with_index { |book, index| puts "#{index}) Title: #{book.title} | Author: #{book.author}" }
+  end
+
+  def list_people
+    @people.each_with_index do |person, index|
+      puts "#{index}) [#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+    end
+  end
+
+  def create_student
+    print 'Age: '
+    age = gets.chomp.to_i
+    print 'Name: '
+    name = gets.chomp
+    print 'Has parent permission? [Y/N]: '
+    parent_permission = gets.chomp.downcase == 'y'
+    print 'Classroom: '
+    classroom = gets.chomp
+    Student.new(age, classroom, name: name, parent_permission: parent_permission)
+  end
+
+  def create_teacher
+    print 'Age: '
+    age = gets.chomp.to_i
+    print 'Name: '
+    name = gets.chomp
+    print 'Specialization: '
+    specialization = gets.chomp
+    Teacher.new(age, specialization, name: name)
+  end
+
+  def create_person
+    puts 'Do you want to create a student (1) or a teacher (2)? [Input the number]: '
+    person_type = gets.chomp.to_i
+
+    if person_type == 1
+      @people << create_student
+      puts 'Student created successfully'
+    elsif person_type == 2
+      @people << create_teacher
+      puts 'Teacher created successfully'
+    else
+      puts 'Invalid option'
+    end
+  end
+
+  def create_book
+    print 'Title: '
+    title = gets.chomp
+    print 'Author: '
+    author = gets.chomp
+    @books << Book.new(title, author)
+    puts 'Book created successfully'
+  end
+
+  def create_rental
+    puts 'Select a book from the following list by number'
+    list_books
+    book_index = gets.chomp.to_i
+    puts 'Select a person from the following list by number (not id)'
+    list_people
+    person_index = gets.chomp.to_i
+    puts 'Date: '
+    date = gets.chomp
+    @rentals << Rental.new(date, @books[book_index], @people[person_index])
```

Comment:
- [ ] Well done your implementation is very good but there is some problem when I enter the `date` it gives an error and the error is `undefined method `add_rental' for nil:NilClass (NoMethodError)` and after that, your app is crashed so kindly fix this error :slightly_smiling_face: 

![Screenshot from 2023-09-02 00-30-57](https://github.com/Gh-Reza/oop-school/assets/45226848/3b31c730-5b9f-446c-8cd3-ecd407a394a9)

Reply:
Hello @aamirkhan2478, thank you for reviewing my code. As I see in the screenshot you selected the person with ID but it should be selected with the index or number - as it is stated in the prompt-.

Reply:
As suggested by the previous reviewer, when a `date` is entered, it throws an `undefined error method` which crashes your application. It would really be nice if you fix the error, as this wasn't selected using an ID as you suggested 😄